src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/setting/handler/ConfigOverridesClusterSettingHandler.java

```diff
@@ -0,0 +1,201 @@
+                .orElseGet(ArrayList::new);
+        List<String> requestedDecidersEnabled = Optional.ofNullable(newEnabled.getDeciders())
+                .orElseGet(ArrayList::new);
+        List<String> requestedDecidersDisabled = Optional.ofNullable(newDisabled.getDeciders())
+                .orElseGet(ArrayList::new);
+
+        List<String> mergedDecidersEnabled = combineLists(currentDecidersEnabled, requestedDecidersEnabled, requestedDecidersDisabled);
+        List<String> mergedDecidersDisabled = combineLists(currentDecidersDisabled, requestedDecidersDisabled, requestedDecidersEnabled);
+
+        merged.getEnable().setDeciders(mergedDecidersEnabled);
+        merged.getDisable().setDeciders(mergedDecidersDisabled);
+    }
+
+    private void mergeActions(final ConfigOverrides merged,
+                              final ConfigOverrides.Overrides baseEnabled,
+                              final ConfigOverrides.Overrides newEnabled,
+                              final ConfigOverrides.Overrides baseDisabled,
+                              final ConfigOverrides.Overrides newDisabled) {
+        List<String> currentActionsEnabled = Optional.ofNullable(baseEnabled.getActions())
+                .orElseGet(ArrayList::new);
+        List<String> currentActionsDisabled = Optional.ofNullable(baseDisabled.getActions())
+                .orElseGet(ArrayList::new);
+        List<String> requestedActionsEnabled = Optional.ofNullable(newEnabled.getActions())
+                .orElseGet(ArrayList::new);
+        List<String> requestedActionsDisabled = Optional.ofNullable(newDisabled.getActions())
+                .orElseGet(ArrayList::new);
+
+        List<String> mergedActionsEnabled = combineLists(currentActionsEnabled, requestedActionsEnabled, requestedActionsDisabled);
+        List<String> mergedActionsDisabled = combineLists(currentActionsDisabled, requestedActionsDisabled, requestedActionsEnabled);
+
+        merged.getEnable().setActions(mergedActionsEnabled);
+        merged.getDisable().setActions(mergedActionsDisabled);
+    }
+
+    /**
+     * Combines three lists by adding all elements in the addList to the base list and
+     * removing all elements in the remove list from the combined list.
+     * // TODO: Add example here to clarify
+     *
+     * @param baseList   The base list.
+     * @param addList    The list whose contents need to added to the base list.
+     * @param removeList The list whose contents should be removed from the base list if present.
+     * @return The combined list as an immutable list.
+     */
+    private List<String> combineLists(List<String> baseList, List<String> addList, List<String> removeList) {
+        Set<String> combinedEnabled = new HashSet<>(baseList);
+        combinedEnabled.addAll(addList);
+        combinedEnabled.removeAll(removeList);
+
+        return ImmutableList.copyOf(combinedEnabled);
```

Comment:
It might be more efficient to use a SetBuilder instead of having to copy the hashSet over again in the end.